Providers/Modules/Plugins/Antimalware/plugin/collectmcafeeinfo.rb

```diff
@@ -9,34 +9,42 @@
 
 class McAfee
 
-	def self.detect()	
-		begin
-			if !File.file?('/opt/isec/ens/threatprevention/bin/isecav')
-				return false
-			end			
-			detectioncmd = `/opt/isec/ens/threatprevention/bin/isecav --version 2>&1`.lines.map(&:chomp)
+	attr_accessor :detectedPath , :mcafeeName , :mcafeeVersion 
+    def self.mcAfeeExist()
+        paths = ['/opt/isec/ens/threatprevention/bin/isecav', '/opt/McAfee/ens/tp/bin/mfetpcli']   
+        for path in paths 
+            if File.file?(path)
+                @detectedPath = path
+                detectioncmd = `#{path} --version 2>&1`.lines.map(&:chomp)
+                # print detectioncmd
+				@mcafeeName = detectioncmd[0]
+                @mcafeeVersion = detectioncmd[1].split(" : ")[1]
```

Comment:
A version was found so return at end of if instead of continuing to check the next path.

Reply:
I am setting the attribute accessor for setting these attributes. So they will only be set when that particular file path is available. 
Hence not returning the attributes here.